system-tests/e2e-suite/src/test/java/com/hotels/styx/E2EJarAccessTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx;
+
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+
+import static com.google.common.base.Throwables.propagate;
+import static com.hotels.styx.ExamplePluginJarLocation.createTemporarySharedDirectoryForJars;
+import static com.hotels.styx.ExamplePluginJarLocation.exampleDependencyJarLocation;
+import static com.hotels.styx.ExamplePluginJarLocation.examplePluginJarLocation;
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static java.nio.file.Files.list;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+// This test is primarily ensuring that our supporting classes and modules are fit-for-purpose
+// i.e. that our example plugin and its dependency are only accessible by legitimately loading their
+// jar files and not just included in the e2e-suite module
+public class E2EJarAccessTest {
```

Comment:
This is not an end to end test at all. Is it?

Reply:
The name comes from the fact that it is checking the behaviour inside the `e2e-suite` module. I am happy to change it to something else.